specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1097 @@
+Subscription) which are expected to conform to the Addressable partial schema
+("duck type"). An object conforms with the Addressable partial schema if it
+contains a `status.address.url` field containing a URL which may be used to
+deliver CloudEvents over HTTP. As a special case, Kubernetes `v1` `Service`
+objects are considered to have a `status.address.url` of
+`http://<service-dns-name>/`. If one of these object references points to an
+object which does not currently satisfy this partial schema (either because the
+`status.address.url` field is empty, or because the object does not have that
+field), then the Trigger or Subscription MUST indicate an error by setting the
+`Ready` condition to `false`, and SHOULD include an indication of the error in a
+condition reason or type.
+
+Both Broker and Channel MUST conform to the Addressable partial schema.
+
+# Event Routing
+
+Note that the event routing description below does not cover the actual
+mechanics of sending an event from one component to another; see
+[the data plane](./data-plane.md) contracts for details of the event transfer
+mechanisms.
+
+## Content Based Routing
+
+A Broker MUST publish a URL at `status.address.uri` when it is able to receive
+events. This URL MUST accept CloudEvents in both the
+[Binary Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[Structured Content Mode](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+HTTP formats. Before sending an HTTP response, the Broker MUST durably enqueue
+the event (be able to deliver with retry without receiving the event again).
+
+For each event received by the Broker, the Broker MUST evaluate each associated
+Trigger **once** (where "associated" means a Trigger with a `spec.broker` which
+references the Broker). If the Trigger has a `Ready` condition of `true` when
+the event is evaluated, the the Broker MUST evaluate the Trigger's `spec.filter`
+and, if matched, proceed with
+[event delivery as described below](#event-delivery). The Broker MAY also
+evaluate and forward events to associated Triggers for which the `Ready`
+condition is not currently `true`. (One example: a Trigger which is in the
+process of being programmed in the Broker data plane might receive _some_ events
+before the data plane programming was complete and the Trigger was updated to
+set the `Ready` condition to `true`.)
+
+If multiple Triggers match an event, one event delivery MUST be generated for
+each match; duplicate matches with the same destination MUST each generate a
+separate event delivery attempts, one per Trigger match. The implementation MAY
+attach additional event attributes or other metaata distinguishing between these
+deliveries. The implementation MUST NOT modify the event payload in this
+process.
+
+Reply events generated during event delivery MUST be re-enqueued by the Broker
```

Comment:
if the re-enqueue fails, the "original" event is treated as failed and will be retried.

Reply:
I think what ville is asking for is what it says a bit down the paragraph: 

> the reply event fails, the entire event delivery MUST be failed and the delivery to the Trigger's subscriber MUST be retried.